### ISSUE TITLE
Add repair panel and section

### ISSUE THREAD

**Issue by asias** [scylladb/scylla-monitoring#1053]
We have the following row level repair metrics:

```
183     row_level_repair_metrics() {
 184         namespace sm = seastar::metrics;
 185         _metrics.add_group("repair", {
 186             sm::make_derive("tx_row_nr", tx_row_nr,
 187                             sm::description("Total number of rows sent on this shard.")),
 188             sm::make_derive("rx_row_nr", rx_row_nr,
 189                             sm::description("Total number of rows received on this shard.")),
 190             sm::make_derive("tx_row_bytes", tx_row_bytes,
 191                             sm::description("Total bytes of rows sent on this shard.")),
 192             sm::make_derive("rx_row_bytes", rx_row_bytes,
 193                             sm::description("Total bytes of rows received on this shard.")),
 194             sm::make_derive("tx_hashes_nr", tx_hashes_nr,
 195                             sm::description("Total number of row hashes sent on this shard.")),
 196             sm::make_derive("rx_hashes_nr", rx_hashes_nr,
 197                             sm::description("Total number of row hashes received on this shard.")),
 198             sm::make_derive("row_from_disk_nr", row_from_disk_nr,
 199                             sm::description("Total number of rows read from disk on this shard.")),
 200             sm::make_derive("row_from_disk_bytes", row_from_disk_bytes,
 201                             sm::description("Total bytes of rows read from disk on this shard.")),
 202         });
 203     }
```
We also have the following progress metrics for repair based node operations and regular repair

```
  58     node_ops_metrics() {
  59         namespace sm = seastar::metrics;
  60         _metrics.add_group("node_maintenance_operations", {
  61             sm::make_gauge("bootstrap_finished_percentage", sm::description("Number of finished percentage for bootstrap operation on this shard."),
  62                 [this] { return bootstrap_finished_percentage(); }),
  63             sm::make_gauge("replace_finished_percentage", sm::description("Number of finished percentage for replace operation on this shard."),
  64                 [this] { return replace_finished_percentage(); }),
  65             sm::make_gauge("rebuild_finished_percentage", sm::description("Number of finished percentage for rebuild operation on this shard."),
  66                 [this] { return rebuild_finished_percentage(); }),
  67             sm::make_gauge("decommission_finished_percentage", sm::description("Number of finished percentage for decommission operation on this shard."),
  68                 [this] { return decommission_finished_percentage(); }),
  69             sm::make_gauge("removenode_finished_percentage", sm::description("Number of finished percentage for removenode operation on this shard."),
  70                 [this] { return removenode_finished_percentage(); }),
  71             sm::make_gauge("repair_finished_percentage", sm::description("Number of finished percentage for repair operation on this shard."),
  72                 [this] { return repair_finished_percentage(); }),
  73         });
  74     }
```

I suggest adding a dedicated repair dash board to display those metrics because 1) it will be too many metrics to add to existing dashboard, e.g, io 2) all of them are useful 3) repair is now the foundation for all the node operations.

**Comment by tzach** [comment]
@amnonh this is for Scylla 4.2

**Comment by amnonh** [comment]
I think that for most users what they want to see is that a repair is currently running and it's progress. I'm not sure that most users would find useful the extra granularity of repair information (disk, network).

For the overview dashboard, I suggest we add:
1. progress bar that indicates a repair progress
2. annotations for the graph.

About the node-operations. @asias who reports that? is that what a node report about itself? or is that the other nodes reporting?

**Comment by asias** [comment]
> I think that for most users what they want to see is that a repair is currently running and it's progress. I'm not sure that most users would find useful the extra granularity of repair information (disk, network).

You can argue that the I/O panel is full of details that most users do not care about, yet the I/O panel is super useful for issue hunting. That's one of the reasons  I recommend adding a dedicated page for repair with all those details. In the overview page, we can add the progress info as you suggested. 

> 
> For the overview dashboard, I suggest we add:
> 
>     1. progress bar that indicates a repair progress
> 
>     2. annotations for the graph.
> 
> 
> About the node-operations. @asias who reports that? is that what a node report about itself? or is that the other nodes reporting?

For regular repair, bootstrap, decommission, replace, rebuild ops, the node running the operation will report the progress. For removenode ops, all nodes that needs sync data will report the progress.

**Comment by amnonh** [comment]
@asias About the remove node, if we are looking to understand that a remove node is in progress, removenode_finished_percentage would probably be initialized to zero, is it set back to zero when done?
So is the global progress: avg(removenode_finished_percentage)/count(removenode_finished_percentage>0) ?

For the self-reporting metrics, is it in sync with the scylla_node_operation_mode?

**Comment by amnonh** [comment]
See also #921

**Comment by amnonh** [comment]
@tzach this is not part of 4.2.rc4, will it be part of 4.2?

**Comment by asias** [comment]
> @asias About the remove node, if we are looking to understand that a remove node is in progress, removenode_finished_percentage would probably be initialized to zero, is it set back to zero when done?
> So is the global progress: avg(removenode_finished_percentage)/count(removenode_finished_percentage>0) ?

Note when no ops in is in  progress or when the ops is finished, the finished_percentage is 1 not 0. We can not turn the percetange to 0 once the ops is finished because the monitor could miss the sampling of the 1. For instance, first sample 98%, next sample is 0%. 

> 
> For the self-reporting metrics, is it in sync with the scylla_node_operation_mode?

They are not completely the same, e.g., there is no replacing operation. The metrics has the name of the operation already. Can we display the operation name from the metrics itself?

```
# HELP scylla_node_operation_mode The operation mode of the current node. UNKNOWN = 0, STARTING = 1, JOINING = 2, NORMAL = 3, LEAVING = 4, DECOMMISSIONED = 5, DRAINING = 6, DRAINED = 7, MOVING = 8
# TYPE scylla_node_operation_mode gauge
scylla_node_operation_mode{shard="0"} 3.000000
```

**Comment by amnonh** [comment]
@asias I'm trying to understand the logic of when to show that something is happening.
So when it's less than 1?

**Comment by asias** [comment]
> @asias I'm trying to understand the logic of when to show that something is happening.
> So when it's less than 1?

Yes, the metric should drop from 1 to zero, then grows to 1.

**Comment by amnonh** [comment]
Depends on https://github.com/scylladb/scylla/issues/7270

**Comment by amnonh** [comment]
Depends on scylladb/scylla#8311

**Comment by amnonh** [comment]
already covered by manager repari
